### PR TITLE
Optimizes Frontend Docker Build

### LINKED ISSUE #914: Optimizes Frontend Docker Builder
In [`docker/Dockerfile-frontend:13`](https://github.com/SCE-Development/Core-v4/blob/dev/docker/Dockerfile-frontend#L13) we are copying the entire contents of Core-v4 into our frontend container. The only directories we really need are `public`, `util` and `src`. Changing the Dockerfile to follow this will decrease the container size and shorten build times.

### PR DESCRIPTION
**Pull request by imdominicreed**
Resolves #914

### CODE REVIEW

**Review by evanugarte — APPROVED** [review]
### Original buildtime:
image

### New buildtime:
image


Over a minute saved! Furthermore if the image is built once then rebuilding it will also save time. For example, here is if I change a file and rebuild:

image

Only 90 seconds! This is a great start, many more revisions to come for sure.

File: docker/Dockerfile-frontend [hunk 1]
```diff
@@ -2,15 +2,13 @@ FROM node:lts-alpine
...
 
-COPY src /usr/src/app/
-COPY public /usr/src/app/
-COPY util /usr/src/app/
-COPY package.json /usr/src/app/
-
 WORKDIR /usr/src/app
-
+COPY package.json ./
 RUN npm install
```
Comment (evanugarte) [inline docker/Dockerfile-frontend:7]:
```suggestion
RUN npm install --production
```

**Review by allen-tran — APPROVED** [review]
oh my gosh this code is so hot 😩💦

**Comment by allen-tran** [comment]
are u single???

**Comment by imdominicreed** [comment]
for you i am 😳

**Pull request merged**